src/fidesops/service/masking/strategy/masking_strategy_aes_encrypt.py

```diff
@@ -1,38 +1,95 @@
-from typing import Optional
+import secrets
+from typing import Optional, List
 
+from fidesops.core.config import config
 from fidesops.schemas.masking.masking_configuration import (
     MaskingConfiguration,
     AesEncryptionMaskingConfiguration,
+    HmacMaskingConfiguration,
 )
+from fidesops.schemas.masking.masking_secrets import MaskingSecretGeneration, SecretType
 from fidesops.schemas.masking.masking_strategy_description import (
     MaskingStrategyDescription,
     MaskingStrategyConfigurationDescription,
 )
 from fidesops.service.masking.strategy.format_preservation import FormatPreservation
 from fidesops.service.masking.strategy.masking_strategy import MaskingStrategy
 from fidesops.util.encryption.aes_gcm_encryption_scheme import encrypt
-
+from fidesops.util.encryption.hmac_encryption_scheme import hmac_encrypt_return_bytes
+from fidesops.util.encryption.secrets_util import SecretsUtil
 
 AES_ENCRYPT = "aes_encrypt"
 
 
 class AesEncryptionMaskingStrategy(MaskingStrategy):
     def __init__(self, configuration: AesEncryptionMaskingConfiguration):
         self.mode = configuration.mode
-        self.key = configuration.key
-        self.nonce = configuration.nonce
         self.format_preservation = configuration.format_preservation
 
-    def mask(self, value: Optional[str]) -> str:
+    def mask(self, value: Optional[str], privacy_request_id: Optional[str]) -> str:
         if self.mode == AesEncryptionMaskingConfiguration.Mode.GCM:
-            masked: str = encrypt(value, self.key, self.nonce)
+            key_hmac: str = SecretsUtil.get_secret(
+                privacy_request_id, AES_ENCRYPT, SecretType.key_hmac
+            )
+            nonce: bytes = self._generate_nonce(value, key_hmac, privacy_request_id)
+            key: bytes = SecretsUtil.get_secret(
+                privacy_request_id, AES_ENCRYPT, SecretType.key
+            )
+            masked: str = encrypt(value, key, nonce)
             if self.format_preservation is not None:
                 formatter = FormatPreservation(self.format_preservation)
                 return formatter.format(masked)
             return masked
         else:
             raise ValueError(f"aes_mode {self.mode} is not supported")
 
+    @staticmethod
+    def _generate_nonce(
+        value: Optional[str], key: str, privacy_request_id: Optional[str]
+    ) -> bytes:
+        salt: str = SecretsUtil.get_secret(
+            privacy_request_id, AES_ENCRYPT, SecretType.salt_hmac
+        )
+        # fixme: replicate what Vault does with sum and remove lower bytes
```

Comment:
Need to research and decide specifics on how we want to sum / trim bytes

Reply:
Looks like `sum()` just computes the checksum of the HMAC strategy, not the sum (mathematical sum) as we  previously thought: Vault uses this hmac crypto package- https://pkg.go.dev/crypto/hmac

Reply:
Ah interesting, that makes sense!